qcengine/config.py

```diff
@@ -72,7 +74,33 @@ class NodeDescriptor(pydantic.BaseModel):
     jobs_per_node: int = 2
     retries: int = 0
 
-    def __init__(self, **data: Dict[str, Any]) -> "BaseModel":
+    # Cluster options
+    is_batch_node: bool = pydantic.Field(False, help="""Whether the node running QCEngine is a batch node
+    
+    Some clusters are configured such that tasks are launched from a special "batch" or "MOM" onto the compute nodes.
+    The compute nodes on such clusters often have a different CPU architecture than the batch nodes and 
+    often are unable to launch MPI tasks, which has two implications:
+        1) QCEngine must make *all* calls to an executable via ``mpirun`` because the executables might not
+        be able to run on the batch node. 
+        2) QCEngine must run on the batch node to be able to launch tasks on the more than one compute nodes  
+    
+    ``is_batch_node`` is used when creating the task configuration as a means of determining whether
+    ``mpirun_command`` must always be used even for serial jobs (e.g., getting the version number)
+    """)
+    mpirun_command: Optional[str] = pydantic.Field(
+        None, description="""Invocation for launching node-parallel tasks with MPI
+        
+        The invocation need not specify the number of nodes, tasks, or cores per node.
+        Information about the task configuration will be added to the command by use of
+        Python's string formatting. The configuration will be supplied as the following variables:
+        
+            {nnodes} - Number of nodes
+            {ranks_per_node} - Number of MPI ranks per node
+            {total_ranks} - Total number of MPI ranks
+        """
+    )
+
+    def __init__(self, **data: Dict[str, Any]):
```

Comment:
Any reason to remove ` -> "BaseModel"` here?

Reply:
Yes, the `__init__` function actually returns `None`